### PR TITLE
TECH-647: Create cucumber test for POST /linked-authorization/link-status

### PR DESCRIPTION
**Pull request by KarolinaKopacz**
Create cucumber test for POST /linked-authorization/link-status

Ticket: https://govstack-global.atlassian.net/browse/TECH-647

### CODE REVIEW

**Comment by KarolinaKopacz** [comment]
Please do not merge before https://github.com/GovStackWorkingGroup/bb-identity/pull/31

File: test/openAPI/features/wallet_generate_link_code.feature [hunk 1]
```diff
@@ -0,0 +1,24 @@
+@method=POST @endpoint=/linked-authorization/link-code
+Feature: The endpoint to generate link code.
+
+  @smoke @unit @positive
+  Scenario: Successfully generate link code smoke type test
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/wallet_generate_link_code.feature:5]:
```suggestion
  Scenario: Successfully generates link code smoke type test
```

File: test/openAPI/features/support/wallet_generate_link_code.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  () =>
+    specWalletGenerateLinkCode
+      .response()
+      .should.have.header(contentTypeHeader.key, contentTypeHeader.value)
+);
+
+Then(
+  'The \\/linked-authorization\\/link-code endpoint response should match json schema',
+  () => {
+    walletGenerateLinkCodeResponseSchema.properties.errors = [];
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/support/wallet_generate_link_code.js:71]: @KarolinaKopacz Are you sure we need this line? Errors ain't required in the schema.

Comment (KarolinaKopacz) [reply test/openAPI/features/support/wallet_generate_link_code.js:71]: Yes, we need it because there should be no errors so array should be empty

Comment (mgrochalskisoldevelo) [reply test/openAPI/features/support/wallet_generate_link_code.js:71]: so should we create a test step that will check if the error array is empty? The schema does not check it atm. I believe it only describes that `errors` property of the response should be an array (if response has the property)

Comment (KarolinaKopacz) [reply test/openAPI/features/support/wallet_generate_link_code.js:71]: This step check it. This is why first I change the `walletGenerateLinkCodeResponseSchema.properties.errors` to empty array and the check if response schema is equals to `walletGenerateLinkCodeResponseSchema`

Comment (KarolinaKopacz) [reply test/openAPI/features/support/wallet_generate_link_code.js:71]: But I will check it again

Comment (mgrochalskisoldevelo) [reply test/openAPI/features/support/wallet_generate_link_code.js:71]: It does not check if the errors array is empty. Try to put 
```
{
      "errorCode": "invalid_transaction_id",
      "errorMessage": "string"
}
```

in the mockoon in the positive response (in errors array) and you will see that all tests succeed and the response still matches with the schema, even if errors array is not empty.

Comment (KarolinaKopacz) [reply test/openAPI/features/support/wallet_generate_link_code.js]: I changed the code a little bit

File: test/openAPI/features/support/wallet_generate_link_code.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+Then(
+  'The \\/linked-authorization\\/link-code endpoint response should match json schema with no errors',
+  () => {
+    chai
+      .expect(specWalletGenerateLinkCode._response.json)
+      .to.be.jsonSchema(walletGenerateLinkCodeResponseSchema);
+    chai
+      .expect(specWalletGenerateLinkCode._response.json.errors)
+      .have.lengthOf(0);
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/support/wallet_generate_link_code.js:76]:
```suggestion
      .to.be.empty;
```

File: test/openAPI/features/support/wallet_list_status.js [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+      .should.have.header(contentTypeHeader.key, contentTypeHeader.value)
+);
+
+Then(
+  'The \\/linked-authorization\\/link-status endpoint response should match json schema with no errors',
+  () => {
+    chai
+      .expect(specWalletLinkStatus._response.json)
+      .to.be.jsonSchema(walletLiskStatusResponseSchema);
+    chai.expect(specWalletLinkStatus._response.json.errors).have.lengthOf(0);
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/support/wallet_list_status.js:97]:
```suggestion
    chai.expect(specWalletLinkStatus._response.json.errors).to.be.empty;
```

**Review by mgrochalskisoldevelo — COMMENTED** [review]
Both ways work, but if `.to.not.be.empty;` is used

I'd suggest to use also
`.to.be.empty;`

instead of 
`.have.langthOf(0);`

for consistency

**Review by mgrochalskisoldevelo — APPROVED** [review]

File: test/openAPI/features/wallet_link_status.feature [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    And The /linked-authorization/link-status endpoint response should be returned in a timely manner 15000ms
+    And The /linked-authorization/link-status endpoint response should have status 200
+    And The /linked-authorization/link-status endpoint response should have content-type: application/json header
+    And The /linked-authorization/link-status endpoint response should match json schema with no errors
+    And The /linked-authorization/link-status response should contain transactionId property equals provided transactionId
+
+  @negative
+  Scenario: Not able to check the status of link code because of the invalid transactionId
+    Given Wants to check the status of link code
+    When Send POST /linked-authorization/link-status request with given X-XSRF-TOKEN header, invalid transactionId, linkCode and requestTime
```
Comment (vishwa-vyom) [inline test/openAPI/features/wallet_link_status.feature:19]: Can we have test scenarios for blank, random, completed transaction id separately ? All will result in invalid_transaction_id error code

File: test/openAPI/features/wallet_link_status.feature [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    And The /linked-authorization/link-status endpoint response should be returned in a timely manner 15000ms
+    And The /linked-authorization/link-status endpoint response should have status 200
+    And The /linked-authorization/link-status endpoint response should have content-type: application/json header
+    And The /linked-authorization/link-status endpoint response should match json schema with errors
+    And The /linked-authorization/link-status response should contain errorCode property equals to "invalid_transaction_id"
+
+  @negative
+  Scenario: Not able to check the status of link code because of the invalid linkCode
+    Given Wants to check the status of link code
+    When Send POST /linked-authorization/link-status request with given X-XSRF-TOKEN header, transactionId, invalid linkCode and requestTime
```
Comment (vishwa-vyom) [inline test/openAPI/features/wallet_link_status.feature:30]: Can we have separate test scenarios for blank, random and link code connected to a different transaction id ? All will result in invalid_link_code error

File: test/openAPI/features/wallet_link_status.feature [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    Then Receive a response from the /linked-authorization/link-status endpoint
+    And The /linked-authorization/link-status endpoint response should be returned in a timely manner 15000ms
+    And The /linked-authorization/link-status endpoint response should have status 200
+    And The /linked-authorization/link-status endpoint response should have content-type: application/json header
+    And The /linked-authorization/link-status endpoint response should match json schema with errors
+    And The /linked-authorization/link-status response should contain errorCode property equals to "invalid_link_code"
```
Comment (vishwa-vyom) [inline test/openAPI/features/wallet_link_status.feature:36]: We can also add one more test scenario where link code is not yet linked and we can expect a response_timeout error code after the 25 seconds

**Review by vishwa-vyom — COMMENTED** [review]
Reviewed feature file and added comments

**Review by vishwa-vyom — COMMENTED** [review]
Reviewed feature file and added comments

File: test/openAPI/features/wallet_link_status.feature (hunk 6 above)
Comment (KarolinaKopacz) [reply test/openAPI/features/wallet_link_status.feature:30]: Sure, I can add scenario with blank and random link code but link code connected to a different transaction id in this case will be the same what random link-code so it return error code

File: test/openAPI/features/wallet_link_status.feature (hunk 7 above)
Comment (KarolinaKopacz) [reply test/openAPI/features/wallet_link_status.feature:36]: For now there is no way to test this case. I will search a solution for it

File: test/openAPI/features/wallet_link_status.feature (hunk 6 above)
Comment (vishwa-vyom) [reply test/openAPI/features/wallet_link_status.feature:30]: the variation here is, it is still a valid link code as per the test application but connected to a different transaction id. We want the test application to ensure this linkage is checked before returning the status.

Comment (vishwa-vyom) [reply test/openAPI/features/wallet_link_status.feature:30]: I see the blank and random link code is available. Are we not taking care of the "link code connected to a different transaction id" ?

File: test/openAPI/features/wallet_link_status.feature (hunk 7 above)
Comment (KarolinaKopacz) [reply test/openAPI/features/wallet_link_status.feature:36]: Whenever the response is returned after more than 25000ms, the test failed. I tried, but we have no option to artificially trigger this error

Comment (vishwa-vyom) [reply test/openAPI/features/wallet_link_status.feature:36]: ok we lets track this test case for future enhancements

**Pull request merged**